microsoft/testsuites/vm_extensions/mdm.py

```diff
@@ -0,0 +1,97 @@
+from pathlib import PurePath
+from typing import Any, Dict, List, cast
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CBLMariner, Debian, Posix, Ubuntu
+from lisa.tools import Curl, Echo, Service
+from lisa.util import LisaException, SkippedException, UnsupportedDistroException
+
+
+@TestSuiteMetadata(
+    area="MetricsExtension",
+    category="functional",
+    description="""
+    This test is a BVT for MDM MetricsExtension
+    """,
+)
+class MetricsExtension(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Verify whether MetricsExtension is installed, running,
+            and uninstalled successfully
+        """,
+        priority=1,
+        use_new_environment=True,
+    )
+    def verify_metricsextension(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+        package = "metricsext2"
+        posix_os: Posix = cast(Posix, node.os)
+        self._is_supported(node)
+
+        # Add repo
+        if isinstance(posix_os, CBLMariner):
+            release = posix_os.information.release
+            curl = node.tools[Curl]
+            curl.fetch(
+                arg="-o /etc/yum.repos.d/mariner-extras.repo",
+                execute_arg="",
+                url=f"https://raw.githubusercontent.com/microsoft/CBL-Mariner/{release}/SPECS/mariner-repos/mariner-extras.repo",  # noqa: E501
+                sudo=True,
+            )
+        else:
+            posix_os.add_azure_core_repo()
+
+        # Install metricsext2
+        is_installed = posix_os.package_exists(package)
+        if not is_installed:
+            if posix_os.is_package_in_repo(package):
+                posix_os.install_packages(package)
+                log.info("MetricsExtension is installed successfully")
+            else:
+                raise LisaException(
+                    f"The supported distro doesn't have {package} in its repo"
+                )
+        else:
+            log.info("MetricsExtension is already installed")
+
+        # Set MDM options
+        mdm_option = variables.get("mdm_option", "")
+        if mdm_option.strip():
+            echo = node.tools[Echo]
+            echo.write_to_file(
+                value=f"{mdm_option}",
+                file=PurePath("/etc/default/mdm"),
+                sudo=True,
+            )
+            log.info("Setting MDM_OPTIONS")
+            log.info(node.execute("cat /etc/default/mdm", sudo=True).stdout)
```

Comment:
The output is already in debug level, it doesn't need to log it again. The info level is to tell a story. Please review all other info log, make sure it's really necessary.

Reply:
Done